language/llama3-405b/SUT_VLLM.py

```diff
@@ -0,0 +1,293 @@
+import asyncio
+import os
+import time
+import numpy as np
+import array
+import torch
+from torch.nn.functional import pad
+from vllm import LLM, AsyncLLMEngine, AsyncEngineArgs, SamplingParams
+from vllm.inputs import TokensPrompt
+
+import pickle
+import time
+import threading
+import tqdm
+import queue
+
+import logging
+from typing import TYPE_CHECKING, Optional, List
+from pathlib import Path
+
+import mlperf_loadgen as lg
+from dataset import Dataset
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("Llama-70B-SUT")
+
+
+class SUT:
+    def __init__(
+        self,
+        model_path=None,
+        dtype="bfloat16",
+        batch_size=None,
+        total_sample_count=24576,
+        dataset_path=None,
+        use_cached_outputs=False,
+        # Set this to True *only for test accuracy runs* in case your prior
+        # session was killed partway through
+        workers=1,
+        tensor_parallel_size=1
+    ):
+
+        self.model_path = model_path or f"Meta-Llama-3.1-405B-Instruct{'-FP8' if dtype == 'float8' else ''}"
+
+        if not batch_size:
+            batch_size = 1
+        self.batch_size = batch_size
+
+        self.dtype = dtype
+        self.tensor_parallel_size = tensor_parallel_size
+
+        if not torch.cuda.is_available():
+            assert False, "torch gpu is not available, exiting..."
+
+        self.dataset_path = dataset_path
+        self.data_object = Dataset(
+            self.model_path,
+            dataset_path=self.dataset_path,
+            total_sample_count=total_sample_count,
+            dtype=dtype
+        )
+        self.qsl = lg.ConstructQSL(
+            self.data_object.total_sample_count,
+            self.data_object.perf_count,
+            self.data_object.LoadSamplesToRam,
+            self.data_object.UnloadSamplesFromRam,
+        )
+
+        self.load_model()
+        gen_kwargs = {
+            "temperature": 1,
+            "top_p": 1,
+            "top_k": 1,
+            "seed": 42,
+            "max_tokens": 256,
```

Comment:
20000

Reply:
Done, I left this accidentally from some tests that I did